modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/api.mustache

```diff
@@ -41,6 +41,9 @@ import {{packageName}}.infrastructure.toMultiValue
     @Deprecated(message = "This operation is deprecated.")
     {{/isDeprecated}}
     {{^doNotUseRxAndCoroutines}}{{#useCoroutines}}suspend {{/useCoroutines}}{{/doNotUseRxAndCoroutines}}fun {{operationId}}({{#allParams}}{{{paramName}}}: {{{dataType}}}{{^required}}?{{/required}}{{^-last}}, {{/-last}}{{/allParams}}) : {{#returnType}}{{{returnType}}}{{#nullableReturnType}}?{{/nullableReturnType}}{{/returnType}}{{^returnType}}Unit{{/returnType}} {
+        {{#isDeprecated}}
+        @Suppress("DEPRECATION")
+        {{/isDeprecated}}
```

Comment:
For the warning, do you mean `This operation is deprecated` (line 41) ?

Reply:
Yes.
Basically in the kotlin generator we have two methods per endpoint.
One that returns the endpoint configuration.
And another that call's the first method to get the endpoint configuration, does the server call and returns the server request.

If this operation is deprecated, both methods will be deprecated.
And when the second method gets the endpoint configuration will show a deprecation warning, but this is just noise for the users of this library.

I'm not sure if I explained it well or not. :)

Reply:
Here are some screenshots to show the issue.

First of all, this change will not impact the users in any way, it's only an internal change.
When the users call a deprecated api, they will still see the deprecated warning, both in the method call and in the request config method.

<img width="389" alt="Screenshot 2021-06-08 at 09 02 30" src="https://user-images.githubusercontent.com/784666/121173678-fe0aed80-c850-11eb-8888-118797b3a18e.png">

The issue that I found and that this PR fixes, is that there are deprecated warnings on the generated code, that don't add any value, and only create noise.


Basically for each api method's, there are two generated methods on the kotlin client.

One that holds the request configurations and only returns it, but it doesn't perform the server request.
<img width="823" alt="Screenshot 2021-06-08 at 09 01 24" src="https://user-images.githubusercontent.com/784666/121174962-6d351180-c852-11eb-98ec-c14fb676c2e2.png">

And another one that actually makes the server request, and get's the configuration of this api method in the previous method.
<img width="1297" alt="Screenshot 2021-06-08 at 09 01 18" src="https://user-images.githubusercontent.com/784666/121174258-a28d2f80-c851-11eb-8cd9-fc0735b70dcd.png">

When an api method is deprecated, both of this methods will be marked as deprecated.

So when the second method that actually makes the server request get's the configuration request, it will show a warning.
But this will only create noise to the user, because this method call is right, and there is nothing to fix here.

That's why I introduced this change.

<img width="1336" alt="Screenshot 2021-06-08 at 09 02 53" src="https://user-images.githubusercontent.com/784666/121174848-4aa2f880-c852-11eb-9d41-380e6c5972bd.png">

Reply:
Thanks for the detailed explanation